components/dropdown/dropdown-content-mixin.js

```diff
@@ -380,14 +386,15 @@ export const DropdownContentMixin = superclass => class extends RtlMixin(supercl
 				this.renderContent = true;
 			}
 
-			doOpen();
+			await doOpen();
 
 		} else {
 
 			if (this.__dismissibleId) {
 				clearDismissible(this.__dismissibleId);
 				this.__dismissibleId = null;
 			}
+			await this.updateComplete;

```

Comment:
Adding an `await` here guarantees that the dropdown will actually be closed when the `d2l-dropdown-close` event is received.